packages/block-editor/src/components/url-input/index.js

```diff
@@ -223,20 +232,63 @@ class URLInput extends Component {
 		this.inputRef.current.focus();
 	}
 
-	static getDerivedStateFromProps( { disableSuggestions }, { showSuggestions } ) {
+	static getDerivedStateFromProps( { value, disableSuggestions }, { showSuggestions, selectedSuggestion } ) {
+		let shouldShowSuggestions = showSuggestions;
+
+		if ( ! value || ! value.length ) {
+			shouldShowSuggestions = false;
+		}
+
+		if ( disableSuggestions === true ) {
+			shouldShowSuggestions = false;
+		}
+
 		return {
-			showSuggestions: disableSuggestions === true ? false : showSuggestions,
+			selectedSuggestion: value && value.length ? selectedSuggestion : null,
+			showSuggestions: shouldShowSuggestions,
 		};
 	}
 
 	render() {
-		const { value = '', autoFocus = true, instanceId, className, id, isFullWidth, hasBorder } = this.props;
-		const { showSuggestions, suggestions, selectedSuggestion, loading } = this.state;
+		const {
+			instanceId,
+			className,
+			id,
+			isFullWidth,
+			hasBorder,
+			renderSuggestions,
+			placeholder = __( 'Paste URL or type to search' ),
+			value = '',
+			autoFocus = true,
+		} = this.props;
```

Comment:
As this is an existing component and already public, lets mark the new props as experimental:
- `renderSuggestions`
- `fetchLinkSuggestions`
- `handleURLSuggestions`
- `onKeyPress`
- `onKeyDown`

I think that's all of them.

Reply:
@marekhrabe Is going to handle all the marking as experimental I believe.

Reply:
we are going to remove `onKeyPress` and `onKeyDown` as well

Reply:
> renderSuggestions
fetchLinkSuggestions
handleURLSuggestions

All done.

> onKeyPress
onKeyDown

No longer in use.